shared/detect-event-type.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/bash
+
+#  Event Types:
+#   - BuildEvent
+#   - MergeEvent
+#   - VerisonEvent
+#   - RebaseEvent
+#
+# The event type is detected based on the action and type of the github-events-resource event
+# These represent the events and types of the events emitted by the github API which are then
+# mapped to the event types above
+
+set -e
+
+ARGV_DIRECTORY="$1"
+
+pushd "$ARGV_DIRECTORY" > /dev/null
+
+action="$(jq -r '.action' .git/.version)"
+type="$(jq -r '.type' .git/.version)"
+
+function emit_build_event() {
```

Comment:
Remove the `function` keyword as is obsolete (https://wiki-dev.bash-hackers.org/scripting/obsolete)